### PR TITLE
Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -55,6 +57,8 @@ jobs:
 
   # Dummy step to gate other tests on without repeating the whole list
   linting-done:
+    permissions:
+      contents: none
```
Comment (reivilibre) [inline .github/workflows/tests.yml:61]: This step doesn't even do anything; what is there to compromise?

Kind of confused about the threat model you have in mind, I guess — what is this protecting against?

**Review by reivilibre — COMMENTED** [review]
I must admit I'm not sure what this is doing or why it's specific to only 2 steps and only the `contents` permission.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .github/workflows/tests.yml:61]: It was good practice to defend any updates in the future. That was the thought process

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -28,6 +28,7 @@ jobs:
       - run: scripts-dev/check_schema_delta.py --force-colors
 
   lint:
+    permissions: {}
     uses: "matrix-org/backend-meta/.github/workflows/python-poetry-ci.yml@v1"
     with:
       typechecking-extras: "all"
```
Comment (DMRobertson) [inline .github/workflows/tests.yml:34]: Does `permissions: {}` even work here? Doesn't the called workflow need permission to at least checkout the repo?

**Review by DMRobertson — CHANGES_REQUESTED** [review]
I don't follow what the intent was here.

Are these changes prompted by some kind of automated tooling?

- Why only limit permissions for the one `lint` job? What about all the others jobs?
- Is there scope to limit permisions granted at the workflow level rather than specific jobs?
- Can you provide a changelog, so CI runs and we can confirm this works?

**Comment by clokep** [comment]
I'm going to close this due to lack of response from the author. If you'd like to continue working on this pull request please respond to the above feedback and let us know!

**Pull request closed without merging**